Added macOS App Bundle soundfont loading

tomkidd opened the pull request:
I'm working on a build of this for Mac Source Ports and I would like it to pull a sound font from the App Bundle, so I added some code to load a .sf2 file. 

If there's a better place or way to do this let me know, this was just a quick and easy way to do it.

1oom-fork commented:
1) SDL_GetBasePath is SDL3. So you're using SDL3 in your build?
2) We have a data folder where we can put lbx. Maybe that would work too. I'll have to think about it.

tomkidd commented:
> 1) SDL_GetBasePath is SDL3. So you're using SDL3 in your build?

No, just SDL2 (SDL_GetBasePath was introduced in SDL2). If SDL1 compatibility is still a goal perhaps this needs to be wrapped in more conditionals. 

> 2) We have a data folder where we can put lbx. Maybe that would work too. I'll have to think about it.

If there's a way to build it with support for in-game music without needing to load in a soundfont, that would work. But if a soundfont is needed I'm a fan of including the file in the bundle to minimize what the end user has to do (i.e., just get the MOO1 data instead of also having to provide their own, etc.)

In that vein, it might be worth adding SDL_GetPrefPath to the sdl2 version of the project to add that to the list of locations for game data to fit the pattern most of the macOS source ports use.

1oom-fork commented:
You're absolutely right. Personally, I'd choose a good GPL-2-licensed soundfont (if such a thing exists) and add it to all platforms at once.

1oom-fork commented:
I think I'll have to first clarify where the LBX files are located, after which it will be possible to search for the soundfont in the same folder. This is possible because LBX is a requirement, not an option.

tomkidd commented:
I've added the ability to load from `~/Library/Application Support/1oom` to fit the pattern a lot of other source ports use on macOS. I added it in such a way that it doesn't conflict with the other Unix data locations.

1oom-fork commented:
Okay. It looks like your approach was the only correct one from the start. We have a hardcoded path for soundfont. We just need to define a default value for Apple, while other systems may not.

1oom-fork commented:
Judging by the description, using the SDL_GetBasePath function is a very specific approach. If we're going with an OS-specific approach, there's no point in using SDL for it, since using SDL is primarily about portability.

1oom-fork commented:
I'm so tired of these pull requests with their history of force pushes.
